### PR TITLE
Increasing timeout values for sockets tests.

### PR DESCRIPTION
**Pull request by CIPop**
Increasing timeout values for sockets tests and attempting to fix protocol issue in UdpClientTests.

#3504, #3641 

@pgavlin @stephentoub @davidsh PTAL

### CODE REVIEW

File: src/System.Net.Sockets.Legacy/tests/FunctionalTests/DnsEndPointTest.cs [hunk 1]
```diff
@@ -159,7 +159,7 @@ public void Socket_ConnectAsyncDnsEndPoint_Success()
             bool willRaiseEvent = sock.ConnectAsync(args);
             if (willRaiseEvent)
             {
-                complete.WaitOne();
+                Assert.True(complete.WaitOne(5000), "Timed out while waiting for connection");
```
Comment (davidsh) [inline src/System.Net.Sockets.Legacy/tests/FunctionalTests/DnsEndPointTest.cs]: I see a lot of literal `5000` numbers.  Would a `const int TimeoutValueInMilliseconds = 5000;` be useful?

**Comment by davidsh** [comment]
Minor comments, otherwise LGTM.

**Comment by pgavlin** [comment]
+1 to David's suggestion re: a named constant. LGTM otherwise.

**Comment by CIPop** [comment]
Thanks for your CR. Will add implementation per #3697 to this PR.

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**